### DISCUSSION TITLE
Create default users on startup

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#8571] [Q&A] [answered]
How can i create 2 default users (Developer, Tester)  while orchard is starting.
I would like to setup predefined role, password email.

**Comment by hishamco** [comment, marked as answer]
You can use Users APIs

**Reply by vitalybrandes** [reply]
Is there any way i can create them on first setup?
Or maybe thru docker variables?

**Reply by hishamco** [reply]
> Is there any way i can create them on first setup?

I think you could check if the tenant is the default one during the setup

**Reply by vitalybrandes** [reply]
I am running only have ne tenant, so yes, but how ?🤔☺️

**Reply by Skrypt** [reply]
I think we are supposed to be able to import/export users now. But they won't export passwords yet.

**Reply by Skrypt** [reply]
So, what @hishamco is specifying is that you can use the IUserService to do so.

**Reply by vitalybrandes** [reply]
Oh, so  i assume i can use some CreateNewUser from IUserService?!
How can i run it once setup complited secsessfully?

**Reply by jtkech** [reply]
Implement an `ISetupEventHandler`, there is one in the `OC.Users` module to create the first administrator user
